docker/Dockerfile

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Base image is the one that is from Jupyter notebook Dockerhub
+# Default environment value is "base", and accepted values are "pyspark" and "tensorflow", for
+# Spark and deep learning environment, respectively.
+ARG ENVIRONMENT='datascience'
+
+FROM jupyter/$ENVIRONMENT-notebook
+LABEL maintainer="Microsoft Recommender Project <RecoDevTeam@service.microsoft.com>"
+
+USER root
+
+# Specify working directory
+ARG WORK_PATH=/home/$NB_USER/work
+ARG RECO_PATH=$WORK_PATH/recommenders
+WORKDIR $WORK_PATH
+
+# Clone the repo
+RUN git clone https://github.com/microsoft/recommenders.git 
+
+# Config conda channel
+RUN conda config --env --add channels conda-forge && \
+    conda config --env --add channels pytorch && \
+    conda config --env --add channels fastai 
+
+# Install the base packages
+RUN conda install --quiet --yes \
+    "mock==2.0.0" \
+    "fastparquet>=0.1.6" \
+    "gitpython>=2.1.8" \
+    "numpy>=1.13.3" \
+    "pymongo>=3.6.1" \
+    "pytest>=3.6.4" \
+    "scikit-surprise>=1.0.6" \
+    "lightgbm==2.2.1" 
+
+RUN pip install \
+    "azureml-sdk[notebooks,contrib]==1.0.18" \
+    "idna==2.7" \
+    "memory-profiler>=0.54.0" \
+    "papermill>=0.15.0" \
+    "pydocumentdb>=2.3.3" \
+    "tqdm==4.31.1"
+
+# Install conda & pip packages
+RUN if ["$ENVIRONMENT" = "pyspark"]; then \
+        # Install the needed packages for pyspark environment
+        conda install --quiet --yes \
+        "pyarrow>=0.8.0" \
+        "pyspark==2.3.1"; \
+    elif ["$ENVIRONMENT" = "tensorflow"]; then \
```

Comment:
maybe for consistency we should say "gpu" instead of  "tensorflow"

Reply:
Hmmmm yea fair point. The reason is that the name of the base image uses `tensorflow` instead of `gpu`. Conditional load of image is a little bit messy in Dockerfile.

Reply:
@miguelgfierro I moved the Dockerfile to the top of the repo, as I noticed that quite a few repos put Dockerfile at the top. E.g., pytorch, horovod, etc.

Reply:
@yueguoguo I didn't come across this situation before.  If you don't like if-else, maybe you can use case:

```
RUN case "$ENVIRONMENT" in \
    "pyspark") \
        # Install the needed packages for pyspark environment
        conda install --quiet --yes \
            "pyarrow>=0.8.0" \
            "pyspark==2.3.1" \
	;; \
    "tensorflow") \
        # Install the needed packages for GPU environment
        conda install --quiet --yes \
            "numba>=0.38.1" \
            "pytorch>=1.0.0" \
            "tensorflow-gpu==1.12.0"
	;; \
    esac
```

But there are still duplicated `conda install --quiet --yes`.

Reply:
> @miguelgfierro I moved the Dockerfile to the top of the repo, as I noticed that quite a few repos put Dockerfile at the top. E.g., pytorch, horovod, etc.

That's true. I also saw Lightgbm does the same. TF, on the contrary has it in an internal folder called [tools](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools). 

I thought that `tools` was the "wild card" to put things that are not tests, utilities and examples. My only concern is that we should be careful not to make the first level of folders too complex. 

But again, up to you, there is more evidence for the first approach